api/src/main/java/com/capitalone/dashboard/webhook/github/GitHubCommitV3.java

```diff
@@ -0,0 +1,318 @@
+        }
+
+        Object repoObject = jsonObject.get("repository");
+        if (!(repoObject instanceof Map)) {
+            return "No Repository Found";
+        }
+
+        repoUrl = restClient.getString(repoObject, "url");
+        String ref = restClient.getString(jsonObject, "ref");
+        if (!StringUtils.isEmpty(ref)) {
+            branch = ref.replace("refs/heads/", "");
+        }
+
+        Object senderObj = jsonObject.get("sender");
+        String authorLDAPDN = restClient.getString(senderObj,"ldap_dn");
+
+        List<Commit> commitList = getCommits(commitsObjectList, repoUrl, branch, authorLDAPDN);
+
+        commitRepository.save(commitList);
+
+        return result;
+    }
+
+    protected List<Commit> getCommits(List<Map> commitsObjectList, String repoUrl,
+                                      String branch, String authorLDAPDN) throws MalformedURLException, HygieiaException, ParseException {
+        List<Commit> commitsList = new ArrayList<>();
+
+        GitHubParsed gitHubParsed = new GitHubParsed(repoUrl);
+        CollectorItem collectorItem = getCollectorItem(gitHubParsed.getUrl(), branch);
+        GitHubWebHookSettings gitHubWebHookSettings = parseAsGitHubWebHook(apiSettings.getGitHubWebHook());
+
+        List<Pattern> commitExclusionPatterns = new ArrayList<>();
+        if (!StringUtils.isEmpty(gitHubWebHookSettings.getNotBuiltCommits())) {
+            if (!StringUtils.isEmpty(gitHubWebHookSettings.getNotBuiltCommitsDelimiter())) {
+                String[] notBuiltCommits = gitHubWebHookSettings.getNotBuiltCommits().split(gitHubWebHookSettings.getNotBuiltCommitsDelimiter());
+                Arrays.stream(notBuiltCommits).map(regExStr -> Pattern.compile(regExStr, Pattern.CASE_INSENSITIVE)).forEach(commitExclusionPatterns::add);
+            } else {
+                commitExclusionPatterns.add(Pattern.compile(gitHubWebHookSettings.getNotBuiltCommits(), Pattern.CASE_INSENSITIVE));
+            }
+        }
+
+        for (Map cObj : commitsObjectList) {
+            Object repoMap = restClient.getAsObject(cObj, "repository");
+            boolean isPrivate = restClient.getBoolean(repoMap, "private");
+            String repoToken = getRepositoryToken(repoUrl);
+            String gitHubWebHookToken =  isPrivate ? repoToken : gitHubWebHookSettings.getToken();
+
+            if (StringUtils.isEmpty(gitHubWebHookToken)) {
+                throw new HygieiaException("Failed processing payload. Missing Github API token in Hygieia.", HygieiaException.INVALID_CONFIGURATION);
+            }
+
+            Commit commit = new Commit();
+
+            String commitId = restClient.getString(cObj, "id");
+            commit.setScmRevisionNumber(commitId);
+
+            Object authorObjectFromCommit = restClient.getAsObject(cObj,"author");
+            String authorName = restClient.getString(authorObjectFromCommit, "name");
+            commit.setScmAuthor(authorName);
```

Comment:
variables authorName, message, commitId can be avoided and directly set into the instance variable commit as not other operation is done on them

Reply:
> variables authorName, message, commitId can be avoided and directly set into the instance variable commit as not other operation is done on them

Cleaned up the variable "authorName", "message" and "commitId" are being used else where, thanks !